config/webpack/plugin/SetupPlugin.js

```diff
@@ -15,13 +46,92 @@ class SetupPlugin {
     const production = compiler.options.mode === 'production';
     const processPath = process.cwd();
 
+    // MODULE
+    // Remove raw markdown rule if found.
+    const mdIndex = compiler.options.module.rules.findIndex((rule) => rule.test.toString && rule.test.toString() === '/\\.md$/');
+    if (mdIndex > -1) {
+      compiler.options.module.rules.splice(mdIndex, 1);
+    }
+
+    const mdxLoader = getMdxLoader(this.publicPath);
+
+    compiler.options.module.rules.push({
+      test: /\.mdx$/,
+      use: [
+        babelLoader,
+        mdxLoader,
+      ],
+    }, {
+      test: /\.md$/,
+      oneOf: [
+        {
+          // To preserve previous behaviour, we will only apply mdx to requests issued
+          // from either the content config files or from .md or .mdx files
+          issuer: [
+            /dev-site-config.*\/contentConfig\.js$/,
+            /\.mdx?$/,
+          ],
+          use: [
+            babelLoader,
+            mdxLoader,
+          ],
+        },
+        {
+          use: 'raw-loader',
+        },
+      ],
+    }, {
+      resourceQuery: /dev-site-codeblock/,
+      // this bypasses the default json loader
+      type: 'javascript/auto',
+      use: [
+        babelLoader,
+        mdxLoader,
+        'devSiteCodeblock',
+      ],
+    }, {
+      resourceQuery: /dev-site-example/,
```

Comment:
Could you explain these? It's out of my webpack knowledge 😬

Reply:
wait is this the backward loader logic explained here? https://webpack.js.org/contribute/writing-a-loader/#complex-usage

Reply:
This is a better source for how to configure loaders with webpack
https://webpack.js.org/configuration/module/#ruleresource
Specifically here we're just testing for the presence of the `?dev-site-example` resource query to run the item through the loaders